components/TextFields/src/MDCTextInputControllerOutlinedField.m

```diff
@@ -116,30 +128,49 @@ - (void)updatePlaceholder {
   [super updatePlaceholder];
 
   CGFloat scale = UIScreen.mainScreen.scale;
-  CGFloat placeholderEstimatedHeight = MDCCeil(self.textInput.placeholderLabel.font.lineHeight * scale) / scale;
+  CGFloat placeholderEstimatedHeight =
+      MDCCeil(self.textInput.placeholderLabel.font.lineHeight * scale) / scale;
   CGFloat placeholderConstant = ([self borderHeight] / 2.f) - (placeholderEstimatedHeight / 2.f);
   if (!self.placeholderCenterY) {
-    self.placeholderCenterY =
-    [NSLayoutConstraint constraintWithItem:self.textInput.placeholderLabel
-                                 attribute:NSLayoutAttributeTop
-                                 relatedBy:NSLayoutRelationEqual
-                                    toItem:self.textInput
-                                 attribute:NSLayoutAttributeTop
-                                multiplier:1
-                                  constant:placeholderConstant];
+    self.placeholderCenterY = [NSLayoutConstraint constraintWithItem:self.textInput.placeholderLabel
+                                                           attribute:NSLayoutAttributeTop
+                                                           relatedBy:NSLayoutRelationEqual
+                                                              toItem:self.textInput
+                                                           attribute:NSLayoutAttributeTop
+                                                          multiplier:1
+                                                            constant:placeholderConstant];
     self.placeholderCenterY.priority = UILayoutPriorityDefaultHigh;
     self.placeholderCenterY.active = YES;
 
     [self.textInput.placeholderLabel setContentHuggingPriority:UILayoutPriorityDefaultHigh + 1
                                                        forAxis:UILayoutConstraintAxisVertical];
   }
   self.placeholderCenterY.constant = placeholderConstant;
+
+  struct MDCSymmetricPadding padding = ((MDCPaddedLabel *)self.textInput.placeholderLabel).padding;
```

Comment:
casting is where things often go wrong.
Please move all casts into there own statement / localvar to ease in parsing.  Here an below.
MDCPaddedLabel *paddedLabel = (MDCPaddedLabel *)self.textInput.placeholderLabel;

Reply:
This is done.